File: app/models/customer_group.py
"""客户群运营领域模型。"""

from dataclasses import dataclass


@dataclass
class CustomerGroup:
    """企业微信客户群绑定信息。"""

    id: str
    chat_id: str
    opengid: str = ""
    name: str = ""
    owner_userid: str = ""
    source: str = ""
    status: str = "active"
    created_at: str = ""
    updated_at: str = ""


@dataclass
class GroupCampaign:
    """客户群团购或预订批次。"""

    id: str
    group_id: str
    title: str
    status: str = "active"
    starts_at: str = ""
    ends_at: str = ""
    summary_note: str = ""
    created_at: str = ""
    updated_at: str = ""


@dataclass
class GroupRegistration:
    """客户提交的群团购登记。"""

    id: str
    campaign_id: str
    group_id: str
    user_id: str
    customer_name: str
    customer_phone: str
    product_name: str
    quantity: int
    fulfillment_method: str = "pickup"
    desired_time: str = ""
    address: str = ""
    remark: str = ""
    status: str = "pending"
    created_at: str = ""
    updated_at: str = ""


__all__ = ["CustomerGroup", "GroupCampaign", "GroupRegistration"]

File: web/admin/src/types/customerGroup.ts
export interface CustomerGroup {
  id: string;
  chatId: string;
  opengid: string;
  name: string;
  ownerUserid: string;
  source: string;
  status: string;
  createdAt: string;
  updatedAt: string;
}

export interface CustomerGroupDraft {
  chatId: string;
  opengid: string;
  name: string;
  ownerUserid: string;
  source: string;
}

export interface GroupCampaign {
  id: string;
  groupId: string;
  title: string;
  status: string;
  startsAt: string;
  endsAt: string;
  summaryNote: string;
  createdAt: string;
  updatedAt: string;
}

export interface GroupCampaignDraft {
  groupId: string;
  title: string;
  startsAt: string;
  endsAt: string;
  summaryNote: string;
}

export interface GroupRegistration {
  id: string;
  campaignId: string;
  groupId: string;
  userId: string;
  customerName: string;
  customerPhone: string;
  productName: string;
  quantity: number;
  fulfillmentMethod: "pickup" | "delivery";
  desiredTime: string;
  address: string;
  remark: string;
  status: "pending" | "confirmed" | "cancelled";
  createdAt: string;
  updatedAt: string;
}

export interface ProductTotal {
  productName: string;
  quantity: number;
}

export interface CampaignSummary {
  campaign: GroupCampaign;
  group: CustomerGroup | null;
  totalRegistrations: number;
  totalQuantity: number;
  statusCounts: Record<string, number>;
  fulfillmentCounts: Record<string, number>;
  productTotals: ProductTotal[];
  pendingFollowups: GroupRegistration[];
  registrations: GroupRegistration[];
  summaryText: string;
}
